schema_editor/test/spec/views/recordtype/related-add-controller.spec.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+describe('ase.views.recordtype: RelatedAddController', function () {
+
+    beforeEach(module('ase.mock.resources'));
+    beforeEach(module('ase.views.recordtype'));
+
+    var $controller;
+    var $httpBackend;
+    var $rootScope;
+    var $scope;
+    var Controller;
+    var ResourcesMock;
+    var StateMock = {
+        go: angular.noop
+    };
+
+    // Initialize the controller and a mock scope
+    beforeEach(inject(function (_$controller_, _$httpBackend_, _$rootScope_,
+                                _ResourcesMock_) {
+        $controller = _$controller_;
+        $httpBackend = _$httpBackend_;
+        $rootScope = _$rootScope_;
+        $scope = $rootScope.$new();
+        ResourcesMock = _ResourcesMock_;
+    }));
+
+    it('should add related content and switch view', function () {
```

Comment:
This test (and some of the others below) could potentially be split into separate tests; it'll require a bit more boilerplate, but I think it's worth it to keep tests as granular as possible since it'll make debugging easier if they break.